kong/cmd/utils/prefix_handler.lua

```diff
@@ -248,6 +248,21 @@ local function compile_conf(kong_config, conf_template, template_env_inject)
     }
   }
 
+  local kong_proxy_access_log = kong_config.proxy_access_log
+  if kong_proxy_access_log ~= "off" then
+    compile_env.proxy_access_log_enabled = true
+  end
+  if kong_proxy_access_log then
+    local _, custom_format = string.match(kong_proxy_access_log, "^(%S+)%s(%S+)")
+    if custom_format then
+      compile_env.proxy_access_log_custom = true
+
+      if not string.match(kong_config.nginx_http_log_format or "", "^" .. custom_format) then
```

Comment:
wouldn't this prevent from using any unescaped magic character in the custom format name, like `test-format`?

Reply:
@samugi make sense.

A format name with a *hyphen* will print the debug message, but does not preventing Kong start. That debug message is a false negative.

Updated with a new commit, so you can verify.

I tested it locally with a real example and it works.

Reply:
@outsinre sorry for a lot of back and forth on this, how about using `string.find` with the `plain` flag set to `true` and look for `"log_format " .. custom_format`?

Reply:
@samugi awesome idea. So we do not have to manipulate the custom format. Let me try.